server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -924,6 +1015,135 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
     }
   }
 
+  private boolean needsFencingForDeletion(MergeInfo info, KeyExtent keyExtent) {
+    return needsFencingForSplit(info, keyExtent) || info.deleteOverlaps(keyExtent);
+  }
+
+  // Instead of splitting or chopping tablets for a delete we instead create ranges
+  // to exclude the portion of the tablet that should be deleted
+  private List<Range> createRangesForDeletion(TabletMetadata tabletMetadata,
+      final KeyExtent deleteRange) {
+    final KeyExtent keyExtent = tabletMetadata.getExtent();
+    final Text prevTabletEndRow =
+        tabletMetadata.getPrevEndRow() != null ? tabletMetadata.getPrevEndRow() : null;
+    final Text tabletEndRow = keyExtent.endRow();
+    final Range tabletRange = tabletMetadata.getExtent().toDataRange();
+    final Key firstKey =
+        prevTabletEndRow != null ? new Key(prevTabletEndRow).followingKey(PartialKey.ROW) : null;
+    final List<Range> ranges = new ArrayList<>();
+
+    // This covers the case of when a deletion range overlaps the first tablet
+    // We need to create a range that excludes the deletion
+    if ((firstKey == null || !firstKey.equals(deleteRange.toDataRange().getStartKey()))
+        && deleteRange.toDataRange().getStartKey() != null
+        && tabletRange.contains(deleteRange.toDataRange().getStartKey())) {
+      Manager.log.trace(
+          "Fencing tablet at start of deletion range with end row {}, Start deletion row {}",
+          prevTabletEndRow, deleteRange.toDataRange().getStartKey().getRow());
+      ranges.add(new Range(prevTabletEndRow, false,
+          deleteRange.toDataRange().getStartKey().getRow(), false));
+    }
+
+    // This covers the case of when a deletion range overlaps the last tablet
+    // We need to create a range that excludes the deletion.
+    if ((tabletEndRow == null
+        || !tabletEndRow.equals(deleteRange.toDataRange().getEndKey().getRow()))
+        && deleteRange.toDataRange().getEndKey() != null
+        && tabletRange.contains(deleteRange.toDataRange().getEndKey())) {
+      Manager.log.trace(
+          "Fencing tablet at end of deletion range with end row {}, End deletion row {}",
+          tabletEndRow, deleteRange.toDataRange().getEndKey().getRow());
+      ranges.add(new Range(deleteRange.toDataRange().getEndKey().getRow(), tabletEndRow));
+    }
```

Comment:
The following does nothing for a null end row and as a result can simplify the code.  Also did not adjust the log message.

```suggestion
    if (deleteRange.endRow() != null && tabletMetadata.getExtent().contains(deleteRange.endRow())) {
      // in the case where the delete range end row is null, there not data to keep in the tablet.
      Manager.log.trace(
          "Fencing tablet at end of deletion range with end row {}, End deletion row {}",
          tabletEndRow, deleteRange.toDataRange().getEndKey().getRow());
      ranges.add(new Range(deleteRange.endRow(), false, tabletEndRow, true));
    }
```

Reply:
I don't think the change here works. All the statements I added in the if clause were necessary (or at least _seemed_ to be but I could be wrong) and done as I found different edge cases when running the DeleteRowsIT test. if you make this change the tests don't work. 

I would need to go and see exactly which of the test cases fail for all the conditions I added as I worked through them one at a time. Specifically for the case of a null endRow, we need to include the range if you are only partially deleting the last tablet. You are right that if the entire last tablet is being deleted then it wouldn't need to be part of a range but if we are only deleting part of the last tablet we need to include it in the range list to be fenced. If you get rid of tabletEndRow == null in the condition then when you run the test it hangs and breaks on this line: https://github.com/apache/accumulo/blob/3409d958bb1d69baf5b93bb8a588a3e3ee9aab1c/test/src/main/java/org/apache/accumulo/test/functional/DeleteRowsIT.java#L111

When I get a chance I can take a look and dig into it more for the different conditions buut the simplest way to verify is to make changes to the logic and then run the tests in `DeleteRowsIT`.

Reply:
ok, I will try running the test and experimenting with the change